### PR TITLE
Add User Group Picker property editor UI

### PR DESCRIPTION
**Pull request by bjarnef**
## Description

In the new backoffice (and the current backoffice) there are property editors for user picker, member picker and member group picker. However I think a user group picker is missing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Motivation and context

A user group picker would make it easier to select specific user groups (and not just a lot of users). It can possible use this in user view similar to how user picker is used on user groups.

Basically what we already have under assign access.

## Screenshots (if appropriate)

image

image

## Checklist

- [ ] If my change requires a change to the documentation, I have updated the documentation in this pull request.
- [x] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/main/.github/CONTRIBUTING.md)>)** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by leekelleher — CHANGES_REQUESTED** [review]
@bjarnef Thank you for the pull request.

As the current backoffice (v13) doesn't have this feature, it hasn't been put on the roadmap for development for v14.

What are your thoughts on developing the User Group picker property-editor itself?

**Comment by bjarnef** [comment]
@leekelleher yes, currently it doesn't exits in Umbraco 13 as a property editor, but it does exists as infinite editor used in user page.
https://github.com/umbraco/Umbraco-CMS/tree/contrib/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/usergrouppicker

However I find it a bit odd we have User Picker, Member Picker and Member Group Picker.. but no User Group Picker.
So I think this is a great chance to implement this.

File: src/packages/core/property-editor/uis/user-group-picker/property-editor-ui-user-group-picker.element.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@customElement('umb-property-editor-ui-user-group-picker')
+export class UmbPropertyEditorUIUserGroupPickerElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	@property()
+	value = '';
+
+	@property({ attribute: false })
+	public config?: UmbPropertyEditorConfigCollection;
+
+	render() {
+		return html`<div>umb-property-editor-ui-user-group-picker</div>`;
```
Comment (leekelleher) [inline src/packages/core/property-editor/uis/user-group-picker/property-editor-ui-user-group-picker.element.ts:19]: The [`<umb-user-group-input>`](https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/v14.0.0--preview004/src/packages/user/user-group/components/input-user-group/user-group-input.element.ts) component could be wired up here.
@bjarnef Would you be up for continuing work on this PR? 😃

Comment (bjarnef) [reply src/packages/core/property-editor/uis/user-group-picker/property-editor-ui-user-group-picker.element.ts:19]: @leekelleher yeah, it was basically a copy of `member-group-picker` here:
https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/9b636d40455f1a015b7de7fe373d063fe11e7ee2/src/packages/core/property-editor/uis/member-group-picker/property-editor-ui-member-group-picker.element.ts#L19

Otherwise feel free to push a change to this PR 😄

Comment (bjarnef) [reply src/packages/core/property-editor/uis/user-group-picker/property-editor-ui-user-group-picker.element.ts]: I have resolved the conflicts and updated to use `<umb-user-group-input>` here instead.

**Comment by iOvergaard** [comment]
I had a look at this again. I think it makes sense to come full circle here, but to my knowledge, we don't have the backend equivalent of this picker. We would need to build the `Umbraco.UserGroupPicker` as a data editor in the backend first.

@bjarnef I will close this PR since we can't merge it right away. You are welcome to create the backend part and get this branch up-to-date with the latest changes, and we'll be able to bring it along for 14.1 for instance 👍
